src/services/services.ts

```diff
@@ -6459,6 +6634,268 @@ namespace ts {
+                return undefined;
+            }
+
+            function filterReferenceEntries(refs: ReferencedSymbol[], filterer: (x: ReferenceEntry) => ReferenceEntry): ReferencedSymbol[] {
+                const result: ReferencedSymbol[] = [];
+                forEach(refs, (ref) => {
+                    const filtered: ReferenceEntry[] = [];
+                    forEach(ref.references, (entry) => {
+                        const filteredEntry = filterer(entry);
+                        if (filteredEntry) {
+                            filtered.push(filteredEntry);
+                        }
+                    });
+                    if (filtered.length) {
+                        result.push({ definition: ref.definition, references: filtered });
+                    }
+                });
+                return result;
+            }
+
+            function filterToClassMemberImplementations(parent: PropertyAccessExpression, searchSymbol: Symbol, refs: ReferencedSymbol[], indexToSymbol: {[index: number]: Symbol}): ReferencedSymbol[] {
+                // Need to find out what class this member is being accessed on
+                const type = typeChecker.getTypeAtLocation(parent.expression);
+                const classHierarchy: Symbol[] = getClassHierarchy(searchSymbol.parent);
+
+                let lowest: ReferencedSymbol[];
+                let lowestRank: number;
+
+                // Filter down the references to those that refer to implementations of the symbol. That is, implementations
+                // from subclasses of the parent class and the lowest implementation in the parent class' class hierarchy
+                forEach(refs, (refSymbol, index) => {
+                    const actual = indexToSymbol[index];
+
+                    // We only care about implementations in classes in the hierarchy
+                    if (actual.parent.getFlags() & SymbolFlags.Interface) {
+                        return;
+                    }
+                    const rank = classHierarchy.indexOf(actual.parent);
+
+                    // No need to check anything past the lowest we have found so far
+                    if (lowest && rank > lowestRank) {
+                        return;
+                    }
+
+                    const implementations: ReferenceEntry[] = [];
+                    forEach(refSymbol.references, (entry) => {
+                        const impl = getImplementationFromEntry(entry, type, classHierarchy);
+                        if (impl) {
+                            implementations.push(impl);
+                        }
+                    });
+                    if (implementations.length) {
+                        if (!lowest || rank < lowestRank) {
+                            lowest = [{ definition: refSymbol.definition, references: implementations }];
+                            lowestRank = rank;
+                        }
+                        else if (rank === lowestRank) {

```

Comment:
i would be interested to know why this happens, i would expect a symbol added only once.